algoliasearch-common/src/main/java/com/algolia/search/AsyncAPIClient.java

```diff
@@ -1357,4 +1359,116 @@ public String generateSecuredApiKey(
+            new AlgoliaRequest<>(
+                HttpMethod.GET,
+                AlgoliaRequestKind.SEARCH_API_READ,
+                Arrays.asList("1", "clusters"),
+                requestOptions,
+                Clusters.class));
+
+    return result.thenApply(Clusters::getClusters);
+  }
+
+  public CompletableFuture<UserIDs> listUserIDs() {
+    return this.listUserIDs(0, 20, new RequestOptions());
+  }
+
+  public CompletableFuture<UserIDs> listUserIDs(Integer page, Integer hitsPerPage) {
+    return this.listUserIDs(page, hitsPerPage, RequestOptions.empty);
+  }
+
+  public CompletableFuture<UserIDs> listUserIDs(
+      @Nonnull Integer page, @Nonnull Integer hitsPerPage, @Nonnull RequestOptions requestOptions) {
+    return httpClient.requestWithRetry(
+        new AlgoliaRequest<>(
+                HttpMethod.GET,
+                AlgoliaRequestKind.SEARCH_API_READ,
+                Arrays.asList("1", "clusters", "mapping"),
+                requestOptions,
+                UserIDs.class)
+            .setParameters(
+                ImmutableMap.of("page", page.toString(), "hitsPerPage", hitsPerPage.toString())));
+  }
+
+  public CompletableFuture<Map<String, List<UserID>>> getTopUserID() {
+    return this.getTopUserID(RequestOptions.empty);
+  }
+
+  public CompletableFuture<Map<String, List<UserID>>> getTopUserID(RequestOptions requestOptions) {
+    CompletableFuture<TopUserResult> result =
+        httpClient.requestWithRetry(
+            new AlgoliaRequest<>(
+                HttpMethod.GET,
+                AlgoliaRequestKind.SEARCH_API_READ,
+                Arrays.asList("1", "clusters", "mapping", "top"),
+                requestOptions,
+                TopUserResult.class));
+
+    return result.thenApply(TopUserResult::getTopUsers);
+  }
+
+  public CompletableFuture<AssignUserID> assignUserID(@Nonnull String userID, @Nonnull String clusterName) {
+    return this.assignUserID(userID, clusterName, RequestOptions.empty);
+  }
+
+  public CompletableFuture<AssignUserID> assignUserID(
+      @Nonnull String userID, @Nonnull String clusterName, RequestOptions requestOptions) {
+    requestOptions.addExtraHeader("X-Algolia-User-ID", userID);
+
+    return httpClient.requestWithRetry(
+        new AlgoliaRequest<>(
+                HttpMethod.POST,
+                AlgoliaRequestKind.SEARCH_API_WRITE,
+                Arrays.asList("1", "clusters", "mapping"),
+                requestOptions,
+                AssignUserID.class)
+            .setData(ImmutableMap.of("cluster", clusterName)));
+  }
+
+  public CompletableFuture<UserID> getUserID(@Nonnull String userID) {
+    return this.getUserID(userID, RequestOptions.empty);
+  }
+
+  public CompletableFuture<UserID> getUserID(
+      @Nonnull String userID, RequestOptions requestOptions) {
+    String encodedUserID = "";
+
+    try {
+      encodedUserID = URLEncoder.encode(userID, "UTF-8");
+    } catch (UnsupportedEncodingException ignore) {
```

Comment:
Same comment as above.

Reply:
Same answer as above ^^

Reply:
And wrapping the error in a `CompletableFuture` as the `getUserID` of the async client cannot throw.